### PR TITLE
Feature/mat 6736 retrieve manifest list

### PR DESCRIPTION
**Pull request by RohitKandimalla**
## Terminology Service PR

Jira Ticket: [MAT-6736](https://jira.cms.gov/browse/MAT-6736)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -30,8 +33,9 @@ spring:
...
   data:
     mongodb:
-      uri: ${MONGO_URI:mongodb://${DBUSER:root}:${DBPASS:E5press0}@localhost:27017/terminology}?authSource=admin&maxPoolSize=50&connectTimeoutMS=2000&serverSelectionTimeoutMS=2000
+      uri: ${MONGO_URI:mongodb://localhost:27017/terminology}?authSource=admin&maxPoolSize=50&connectTimeoutMS=2000&serverSelectionTimeoutMS=2000
 
 mapping:
   data:
-    code-system-entry-url: ${CODE_SYSTEM_ENTRY_URL:https://madie-dev-static.s3.amazonaws.com/mappings/code-system-entry.json}
+    code-system-entry-url: file:/Users/43093/Documents/git/madie-ms/support-data/prod/madie/code-system-entry.json
+#    code-system-entry-url: ${CODE_SYSTEM_ENTRY_URL:https://madie-dev-static.s3.amazonaws.com/mappings/code-system-entry.json}
```
Comment (adongare) [inline src/main/resources/application.yml]: looks like local change. need to be reverted?

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -30,8 +33,9 @@ spring:
     max-in-memory-size: 24MB
   data:
     mongodb:
-      uri: ${MONGO_URI:mongodb://${DBUSER:root}:${DBPASS:E5press0}@localhost:27017/terminology}?authSource=admin&maxPoolSize=50&connectTimeoutMS=2000&serverSelectionTimeoutMS=2000
+      uri: ${MONGO_URI:mongodb://localhost:27017/terminology}?authSource=admin&maxPoolSize=50&connectTimeoutMS=2000&serverSelectionTimeoutMS=2000
```
Comment (adongare) [inline src/main/resources/application.yml]: this as well

File: src/main/java/gov/cms/madie/terminology/service/FhirTerminologyService.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class FhirTerminologyService {
+  FhirContext ctx = FhirContext.forR4();
```
Comment (adongare) [inline src/main/java/gov/cms/madie/terminology/service/FhirTerminologyService.java]: there is already a FHIRContext available in service. Can be injected it here.

File: src/main/resources/application.yml (hunk 1 above)
Comment (RohitKandimalla) [reply src/main/resources/application.yml]: ooh shoot, I will revert these

File: src/main/java/gov/cms/madie/terminology/service/FhirTerminologyService.java (hunk 3 above)
Comment (RohitKandimalla) [reply src/main/java/gov/cms/madie/terminology/service/FhirTerminologyService.java]: Yeah I have seen that, But I am not sure why we have to go through an extra step of having a new Bean for FhirContext when we can directly access forR4() which is a static method.

Comment (adongare) [reply src/main/java/gov/cms/madie/terminology/service/FhirTerminologyService.java]: that is because as per FHIRCOntext documentation it is lil bit heavy operation

> The FHIR context is the central starting point for the use of the HAPI FHIR API. It should be created once, and then used as a factory for various other types of objects (parsers, clients, etc.).

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -22,6 +22,9 @@ client:
   vsac_base_url: https://vsac.nlm.nih.gov/vsac
   valueset_endpoint: /svs/RetrieveMultipleValueSets?id={oid}&profile={profile}&includeDraft={includeDraft}
   default_profile: ${DEFAULT_VSAC_PROFILE:Most Recent Code System Versions in VSAC}
+  fhir-terminology-service:
+    base-url: https://uat-cts.nlm.nih.gov/fhir
```
Comment (adongare) [inline src/main/resources/application.yml]: this url seems temporary. should we introduce env var for this?

File: src/main/java/gov/cms/madie/terminology/service/FhirTerminologyService.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  private final FhirTerminologyServiceWebClient fhirTerminologyServiceWebClient;
+
+  @Cacheable("manifest-list")
```
Comment (adongare) [inline src/main/java/gov/cms/madie/terminology/service/FhirTerminologyService.java:24]: do we need to have time to live for cache or is there a default ttl by spring/boot? I don't think it changes often but just thinking

Comment (RohitKandimalla) [reply src/main/java/gov/cms/madie/terminology/service/FhirTerminologyService.java:24]: Yeah I am not sure either, but to force refresh the list ( which is expected not to change often ) I thought we could just restart our service

File: src/main/resources/application.yml (hunk 4 above)
Comment (RohitKandimalla) [reply src/main/resources/application.yml]: Yes, good catch, I added a new variable. Will keep Brandon updated

File: src/main/java/gov/cms/madie/terminology/service/FhirTerminologyService.java (hunk 3 above)
Comment (RohitKandimalla) [reply src/main/java/gov/cms/madie/terminology/service/FhirTerminologyService.java]: Cool, Good to know, and I updated it to inject the bean.

**Review by adongare — DISMISSED** [review]
LGTM

**Review by adongare — APPROVED** [review]
